coldfront/core/utils/management/commands/audit_data.py

```diff
@@ -0,0 +1,229 @@
+                    self.stdout.write(self.style.ERROR('Allocation {} for project {}'
+                    ' has an end date while being an inactive ICA.' \
+                        .format(id, project)))
+                if project_status == 'Active' and not any(end_date != ica.end_date for ica in ICA_ALLOCATION_PERIODS):
+                    self.stdout.write(self.style.ERROR('Allocation {} for project {}'
+                    ' is an active ICA that has an end date that is different from all ICA allocation periods.' \
+                        .format(id, project)))
+            elif project.startswith(RECHARGE_CONDO_PREFIX):
+                if end_date is not None:
+                    self.stdout.write(self.style.ERROR('Allocation {} for project {}'
+                    ' is a Recharge or Condo allocation with an end date.' \
+                        .format(id, project)))
+    
+    def handle_project_inactive(self):
+        '''
+        Assert that inactive Projects have "Expired" allocations and zero SUs.
+        '''
+        inactive_projects = Project.objects.prefetch_related('allocation_set__status').filter(status__name='Inactive') 
+        
+        for project in inactive_projects:
+            for allocation in project.allocation_set.select_related('status').prefetch_related('allocationattribute_set'):
+                if allocation.status.name != 'Expired':
+                    self.stdout.write(self.style.ERROR('Project {} ({})'
+                    ' is inactive and has unexpired allocations.' \
+                        .format(project.id, project.name)))
+                try:
+                    allocation_attribute = allocation.allocationattribute_set.get(allocation_attribute_type__name='Service Units')
+                    if allocation_attribute.value == 0:
+                        self.stdout.write(self.style.ERROR('Project {} ({})'
+                    ' is inactive and has non-zero SUs.' \
+                            .format(project.id, project.name)))
+                except AllocationAttribute.DoesNotExist as e:
+                    pass
+
+    def handle_project_pi(self):
+        '''
+        Assert that all projects have at least one PI.
+        '''
+        projects = Project.objects.all()
+        for project in projects:
+            if project.pis().count() == 0:
+                self.stdout.write(self.style.ERROR('Project {} ({})'
+                    ' has no PIs.' \
+                    .format(project.id, project.name)))
+
+    def handle_user_project(self):
+        '''
+        Assert that all users with a cluster UID should be associated with at least one Project.
+        '''
+        users = UserProfile.objects.filter(cluster_uid__isnull=False) \
+            .values('id', 'user__first_name', 'user__last_name', 'user__email', 'access_agreement_signed_date')
+        for user in users:
+            user_project_count = ProjectUser.objects.filter(user_id=user['id']).count()
```

Comment:
We can filter to only active users, yeah. Maybe we should also filter to users that have signed the access agreement as they make up all the output of the command (looks like all inactive users haven't signed, but some active users haven't also signed either). What do you think?

Reply:
I'm not sure about the access agreement. Lets leave that up to Matthew

Reply:
I just found out that since denied and removed projectusers still have an entry in the projectuser table, the exists() method returns true for them, so they were excluded by default.

Reply:
We can ignore whether they've signed the access agreement.

When we initially loaded in users, we made everyone "Active", but left the agreement unsigned.

Anyone coming to the portal needs to sign it as the first step. But there may still be some active cluster users who haven't signed the agreement because they haven't needed to come to the portal yet.

Reply:
So does that mean no changes needed then?